packages/kotti-ui/source/kotti-comment/components/CommentInlineEdit.vue

```diff
@@ -1,78 +1,82 @@
 <template>
-	<div v-if="!isEditing" class="kt-comment-inline-edit__message">
-		<!-- eslint-disable vue/no-v-html -->
-		<span v-html="postEscapeParser(dangerouslyOverrideParser(message))" />
-		<!-- eslint-enable vue/no-v-html -->
-	</div>
-	<div v-else class="kt-comment-inline-edit">
-		<textarea v-model="inlineValue" class="form-input" />
-		<KtButtonGroup class="kt-comment-inline-edit__buttons">
-			<KtButton icon="close" @click="handleCancel" />
-			<KtButton icon="check" @click="handleConfirm" />
-		</KtButtonGroup>
+	<div class="kt-comment__inline-edit">
+		<CommentTextArea
+			v-if="isEditing"
+			v-model="_message"
+			:allowInternal="allowInternal"
+			autofocus
+			:isInternal="isInternal"
+			:isReply="isReply"
+			:tabIndex="tabIndex"
+			@cancel="onCancel"
+			@confirm="onConfirm"
+			@toggleInternal="onToggleInternal"
+		/>
+		<div v-else class="kt-comment__inline-edit__message">
+			<!-- eslint-disable vue/no-v-html -->
+			<span v-html="postEscapeParser(dangerouslyOverrideParser(message))" />
+			<!-- eslint-enable vue/no-v-html -->
+		</div>
 	</div>
 </template>
 
 <script lang="ts">
 import { defineComponent, ref, watch } from '@vue/composition-api'
 
-import { KtButton } from '../../kotti-button'
-import { KtButtonGroup } from '../../kotti-button-group'
+import { makeProps } from '../../make-props'
 import { KottiComment } from '../types'
 
-export default defineComponent<{
-	dangerouslyOverrideParser: () => void
-	id: KottiComment.PropsInternal['id']
-	isEditing: boolean
-	message: string
-	parentId?: KottiComment.PropsInternal['id']
-	postEscapeParser: () => void
-}>({
+import CommentTextArea from './CommentTextArea.vue'
+
+export default defineComponent<KottiComment.InlineEdit.PropsInternal>({
 	name: 'CommentInlineEdit',
 	components: {
-		KtButton,
-		KtButtonGroup,
-	},
-	props: {
-		dangerouslyOverrideParser: { required: true, type: Function },
-		id: { required: true, type: [Number, String] },
-		isEditing: { default: false, type: Boolean },
-		message: { required: true, type: String },
-		parentId: { required: false, type: [Number, String] },
-		postEscapeParser: { required: true, type: Function },
+		CommentTextArea,
 	},
+	props: makeProps(KottiComment.InlineEdit.schema),
 	setup(props, { emit }) {
-		const inlineValue = ref<string | null>(null)
+		const _message = ref<KottiComment.InlineEdit.PropsInternal['message']>('')
 
 		watch(
 			() => props.isEditing,
-			(newValue) => {
-				if (newValue === true) {
-					inlineValue.value = props.message
+			(isEditing) => {
+				if (isEditing) {
+					_message.value = props.message
```

Comment:
there's a caveat

if there's a user scenario that would allow the user somehow to trigger the update:isEditing to true twice in a row
then the message will get reset

imagine he clicks post (isEditing is set to true, watcher triggers, so you have your local value set)

then user starts typing

if by any mean, they manage to retrigger update:isEditing then you'd have a bug of resetting your local variable mid typing

and it'd be avoided with a check
```
if(isEditing === props.isEditing) return 
```

Reply:
if it's not possible to end up in this scenario, ignore

Reply:
It's not really necessary bc the only way to trigger it is via the `Edit` button inside `Actions`, but it's hidden during editing.

Nevertheless, I applied the changed to make it explicit the the internal value shouldn't be mutated during editing.